routes/channel.js

```diff
@@ -92,29 +69,70 @@ function getList(req, res, next) {
 		.catch(next)
 }
 
-// Implementation of getValidatorMessagesDetailed
-// It retrieves the last recent N
-// validator messages
+// Implementation of getValidatorMessages
+// It retrieves the last N validator messages
+const VALIDATOR_MSGS_PROJ = { _id: 0, channelId: 0 }
 function getValidatorMessages(req, res, next) {
 	const { type, id, uid } = req.params
 	const { limit } = req.query
 
-	const validatorCol = db.getMongo().collection('validatorMessages')
+	const validatorMsgCol = db.getMongo().collection('validatorMessages')
 	const query = { channelId: id }
 	if (typeof uid === 'string') query.from = uid
-	if (typeof type === 'string') query['msg.type'] = type
+	if (typeof type === 'string') {
+		query['msg.type'] = { $in: type.split('+') }
+	}
 
-	validatorCol
-		.find(query)
-		.sort({ _id: -1 })
+	validatorMsgCol
+		.find(query, { projection: VALIDATOR_MSGS_PROJ })
+		.sort({ received: -1 })
 		.limit(limit ? Math.min(cfg.MSGS_FIND_LIMIT, limit) : cfg.MSGS_FIND_LIMIT)
 		.toArray()
 		.then(function(validatorMessages) {
-			res.send({ validatorMessages, channel: req.channel })
+			res.send({ validatorMessages })
 		})
 		.catch(next)
 }
 
+function getLastApprovedMessages(req, res, next) {
+	retrieveLastApproved(req.channel)
+		.then(lastApproved => res.send({ lastApproved }))
+		.catch(next)
+}
+
+async function retrieveLastApproved(channel) {
+	const validatorMsgCol = db.getMongo().collection('validatorMessages')
+	const approveStateMsgs = await validatorMsgCol
+		.find(
+			{
+				channelId: channel.id,
+				from: channel.validators[1],
+				'msg.type': 'ApproveState'
+			},
+			{ projection: VALIDATOR_MSGS_PROJ }
+		)
+		.sort({ received: -1 })
+		.limit(1)
+		.toArray()
+	if (!approveStateMsgs.length) {
+		return null
```

Comment:
Empty `[]` might be more appropiate instead of `null`?

Reply:
the response for success is `{ newState, approveState }`, this is why a `null` is better; also, if the response is non-null, there's always `{newState, approveState}`, which makes the type more correct cause it's either nothing, or a valid last approved pair